spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/CompletionController.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.admin.controller;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.dataflow.completion.CompletionProposal;
+import org.springframework.cloud.dataflow.completion.StreamCompletionProvider;
+import org.springframework.cloud.dataflow.core.StreamDefinition;
+import org.springframework.cloud.dataflow.rest.resource.CompletionProposalsResource;
+import org.springframework.cloud.dataflow.rest.resource.StreamDefinitionResource;
+import org.springframework.hateoas.ExposesResourceFor;
+import org.springframework.hateoas.mvc.ResourceAssemblerSupport;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Exposes the DSL completion features of CompletionProvider as a REST API.
+ *
+ * @author Eric Bottard
+ */
+@RestController
+@RequestMapping("/completions")
+@ExposesResourceFor(CompletionProposalsResource.class)
+public class CompletionController {
+
+	private final StreamCompletionProvider completionProvider;
+
+	private Assembler assembler = new Assembler();
+
+	@Autowired
+	public CompletionController(StreamCompletionProvider completionProvider) {
+		this.completionProvider = completionProvider;
+	}
+
+	/**
+	 * Return a list of possible completions given a prefix string that the user has started typing.
+	 *  @param start the amount of text written so far
+	 * @param detailLevel the level of detail the user wants in completions
```

Comment:
Is there an accepted range of values for `detailLevel`? This should be specified here and in other places where it is accepted as a parameter.